nyan/value/value_holder.h

```diff
@@ -74,6 +92,20 @@ class ValueHolder {
 	std::shared_ptr<Value> value;
 };
 
+
+template <ValueLike T>
+const std::shared_ptr<T> ValueHolder::get_value_ptr() const {
+	auto ret = std::dynamic_pointer_cast<T>(this->value);
+
+	if (not ret) {
+		throw InternalError{"ValueHolder does not contain a value of type "
```

Comment:
i think this shouldn't be a internal error since the api user can specify a wrong type and then "intentionally" get a wrong value which won't cast. internal errors are for internal constraint checks, this is rather an api/object model usage error